marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -420,12 +420,28 @@ handleAction s (NewProjectAction action@(NewProject.CreateProject lang)) = do
   assign _gistId Nothing
   assign _createGistResult NotAsked
   liftEffect $ LocalStorage.setItem gistIdLocalStorageKey mempty
-  toHaskellEditor $ HaskellEditor.editorSetValue HE.example
-  liftEffect $ LocalStorage.setItem bufferLocalStorageKey HE.example
-  for_ (Map.lookup "Example" StaticData.demoFilesJS) \contents -> void $ query _jsEditorSlot unit (Monaco.SetText contents unit)
-  liftEffect $ LocalStorage.setItem jsBufferLocalStorageKey JE.example
-  toSimulation $ Simulation.editorSetValue "?new_contract"
-  liftEffect $ LocalStorage.setItem marloweBufferLocalStorageKey "?new_contract"
+  -- reset all the editors
+  toHaskellEditor $ HaskellEditor.editorSetValue mempty
+  liftEffect $ LocalStorage.setItem bufferLocalStorageKey mempty
+  void $ query _jsEditorSlot unit (Monaco.SetText mempty unit)
+  liftEffect $ LocalStorage.setItem jsBufferLocalStorageKey mempty
+  toSimulation $ Simulation.editorSetValue mempty
+  liftEffect $ LocalStorage.setItem marloweBufferLocalStorageKey mempty
+  void $ query _blocklySlot unit (Blockly.SetCode mempty unit)
+  -- set the appropriate editor
+  case lang of
+    Haskell ->
+      for_ (Map.lookup "Example" StaticData.demoFiles) \contents -> do
+        toHaskellEditor $ HaskellEditor.editorSetValue HE.example
+        liftEffect $ LocalStorage.setItem bufferLocalStorageKey HE.example
+    Javascript ->
+      for_ (Map.lookup "Example" StaticData.demoFilesJS) \contents -> do
+        void $ query _jsEditorSlot unit (Monaco.SetText contents unit)
```

Comment:
Why is it that the JS editor gets accessed directly, when the Haskell/Marlowe ones are wrapped?

Reply:
~laziness~ the js editor is in need of a good refactor anyway. When you have this in your mind you become susceptible to "I'll change it when I do the other refactoring" despite being exactly what you should _not_ be thinking!